doc/threat_model/README.md

```diff
@@ -0,0 +1,158 @@
+side channels that Tock can mitigate in theory are too expensive for Tock to
+mitigate in practice. As a result, Tock does not mitigate side channels by
+default. However, specific Tock components may provide and document their own
+side channel mitigation. For instance, Tock may provide a cryptography API that
+implements constant-time operations, and may document the side channel defense
+in the cryptography API's documentation.
+
+In deciding whether to mitigate a side channel, Tock developers should consider
+both the cost of mitigating the side channel as well as the value provided by
+mitigating that side channel. For example:
+
+1. Tock does not hide an application's CPU usage from other applications. Hiding
+   CPU utilization generally requires making significant performance tradeoffs,
+   and CPU utilization is not a particularly sensitive signal.
+
+1. Although Tock protects an application's data from unauthorized access, Tock
+   does not hide the size of an application's data regions. Without virtual
+   memory hardware, it is very difficult to hide an application's size, and that
+   size is not particularly sensitive.
+
+1. It is often practical to build constant-time cryptographic API
+   implementations, and protecting the secrecy of plaintext is valuable. As
+   such, it may make sense for a Tock board to expose a cryptographic API with
+   some side channel defenses.
+
+### Shared Buses
+
+External communication buses (for instance, an internet connection or a UART)
+may be shared between applications. When buses are shared between applications,
+Tock should provide isolation commensurate with the isolation provided by the
+underlying communication technology. For example:
+
+1. A UDP API can provide a mechanism for clients (applications and/or untrusted
+   capsules) to gain exclusive access to a port. The UDP API should then prevent
+   clients from reading messages sent to other clients or impersonating other
+   clients.
+
+1. A UART API with multiple clients cannot determine which client owns received
+   data. Tock is unable to prevent one application from reading UART data meant
+   for a different application.
+
+## Components Trusted to Provide Isolation
+
+The Tock kernel depends on several components (including hardware and software)
+in order to implement the above isolation guarantees. Some of these components,
+such as the application loader, may vary depending on Tock's use case. The
+following documents describe the trust model that exists between the Tock kernel
+and its security-relevant dependencies:
+
+- [Untrusted Capsule Isolation](Untrusted_Capsule_Isolation.md) describes the
+  coding practices used to isolate untrusted capsules from the remainder of the
+  kernel.
+
+- [Application Loader](Application_Loader.md) describes the trust placed in the
+  application deployment mechanism.
+
+- [TBF Headers](TBF_Headers.md) describes the trust model associated with the
```

Comment:
You should define what "TBF" means here.

Reply:
Done.